### DISCUSSION TITLE
GeoTIFF create by prod.stack is not readable by QGIS

### DISCUSSION THREAD

**Discussion by homoludens** [sertit/eoreader#116] [Q&A] [answered]
I am using Sentinel-2 images, loading and using different bands and indices is working, but I want to save GeoTIFF and open it in QGIS. As I have found this comment: https://gis.stackexchange.com/a/448194:

`prod.stack([GREEN], stack_path="/tmp/green.tif")`

But QGIS reports: `Invalid Data Source: green.tif is not a valid or recognized data source.`

GIMP can open it, rasterio too, thou I need to convert to uint8 to make it usable.  Basically I can save it, use rasterio to open and use rasterio to get what I need, but seams too convoluted. 

But, do you have any idea how to discover what is the problem or even better how to create GeoTIFF compatible with QGIS?

**Comment by remi-braun** [comment]
Hello,

This is strange, all GTiff should be QGIS-compatible.
Could you send me this `green.tif` ?
Is there any warning or something when saving your stack on disk ?

What are your versions of:
- eoreader
- python
- QGIS
---
Note that if you set an output folder somewhere, you should be able to access intermediary files : all the spectral band, spectral indices etc. 

```python
prod = reader.open(path, output_path="/your/path/")
green = prod.load([GREEN])["GREEN"]
```

You'll have a green band in `/your/path/`

**Reply by homoludens** [reply, marked as answer]
While preparing answer for you and trying out `.open(path, output_path="/your/path/")` I discovered that problem is me using `/tmp` directory, looks like file in it is just link to some subfolder FlatPak QGIS does not have access too.  It is opening tif files nicely from different folder. 

Sorry for making noise, but I learned useful thing about this output_path. I haven't seen (or understand) that from documentation. I guess some more basic examples would be nice to have :). 


Thank you for your help!


--------
Here is the answer to your questions, even thou it is not needed:

python-3.11
eoreader==0.20.4
QGIS 3.32.2-Lima (Linux, FlatPak)

Sorry for wetransfer link: https://we.tl/t-u45P8l6fZT but file is 447MB, I made no additional manipulation but:

```
from eoreader.reader import Reader
from eoreader.bands import GREEN, NDVI, CLOUDS, NDWI, RED, BLUE
prod = Reader().open('S2B_MSIL2A_20210218T094029_N0214_R036_T34TDR_20210218T123009.zip')
prod.stack(GREEN, stack_path="/tmp/green.tif")
```

**Reply by remi-braun** [reply]
Yes you are right, I just saw I haven't documented it properly in the current notebooks.
I added an issue about that, thank you for the feedback :) #117
